src/offlinelogstorage/dlt_offline_logstorage_behavior.c

```diff
@@ -732,6 +736,29 @@ int dlt_logstorage_open_log_file(DltLogStorageFilterConfig *config,
                 }
             }
 
+#ifdef DLT_LOGSTORAGE_USE_GZIP
+            if (config->gzip_compression == DLT_LOGSTORAGE_GZIP_FILE) {
+                tmp = &config->records;
+                while ((*tmp)->next != NULL) {
+                    int len = strlen((*tmp)->name);
+                    const char *suffix = ".dlt";
```

Comment:
I need more context/info/description about this feature:
1. Is it trying to do a dynamic activate compress mode, e.g. logging .dlt and then immediately switch to .dlt.gz using compress function for file/fd?
2. If it does so, why dont we just apply compress function in native logstorage file rotation, adding new func here seems doing a duplicated task?
3. Minor suggestion: adding macro name FILE cannot directly inform user/dev that it is for file rotation, any better naming scheme here?

Reply:
1. The feature it to log normal dlt and only when a file is "full" it compress it. This way the stored files are compressed but the current file is uncompressed. This is for the use case of copying stored files while logging is in progress.
I think the purpose of removing older files and having a max number of files is to not fill storage. Then it would make sense to bunch dlt and gz.dlt together into one count.

2. It is quite different tasks to compress into an open file and to compress an entire file. I thought this way was the easiest and cleanest way.

3. I thought of it but couldn't figure out a better name that is nice and short. Maybe ARCHIVE is better?